src/phoenix/auth.py

```diff
@@ -33,16 +61,165 @@ def validate_password_format(password: str) -> None:
+        if any(char.isspace() for char in password):
+            raise ValueError("Password must not contain whitespace characters")
+        if not password.isascii():
+            raise ValueError("Password must contain only ASCII characters")
+        err_msg = []
+        if len(password) < self.length:
+            err_msg.append(f"must be at least {self.length} characters long")
+        if self.special_chars and not any(char in "!@#$%^&*()_+" for char in password):
+            err_msg.append("at least one special character")
+        if self.digits and not any(char.isdigit() for char in password):
+            err_msg.append("at least one digit")
+        if self.upper_case and not any(char.isupper() for char in password):
+            err_msg.append("at least one uppercase letter")
+        if self.lower_case and not any(char.islower() for char in password):
+            err_msg.append("at least one lowercase letter")
+        if not err_msg:
+            return
+        try:
+            from faker import Faker  # type: ignore[import-not-found,unused-ignore]
+
+            suggested_password = Faker().unique.password(**asdict(self))
+        except BaseException:
+            suggested_password = ""
+        if len(err_msg) > 1:
+            err_text = "Password " + ", ".join(err_msg[:-1]) + ", and " + err_msg[-1]
+        else:
+            err_text = f"Password {err_msg[0]}"
+        if suggested_password:
+            err_text = f"{err_text} (e.g. `{suggested_password}`)"
+        raise ValueError(err_text)
+
+
+"""The name of the header for token-based authentication"""
 EMAIL_PATTERN = re.compile(r"^[^@\s]+@[^@\s]+[.][^@\s]+\Z")
+"""The regular expression pattern for a valid email address."""
 NUM_ITERATIONS = 10_000
+"""The number of iterations to use for the PBKDF2 key derivation function."""
 MIN_PASSWORD_LENGTH = 4
+"""The minimum length of a password."""
+PASSWORD_REQUIREMENTS = _PasswordRequirements(MIN_PASSWORD_LENGTH)
+"""The requirements for a valid password."""
+REQUIREMENTS_FOR_PHOENIX_SECRET = _PasswordRequirements(32, True, True, True, True)
+"""The requirements for the Phoenix secret key."""
+JWT_ALGORITHM = "HS256"
+"""The algorithm to use for the JSON Web Token."""
+PHOENIX_ACCESS_TOKEN_COOKIE_NAME = "phoenix-access-token"
+PHOENIX_ACCESS_TOKEN_COOKIE_MAX_AGE = timedelta(days=31)
+
+
+class Issuer(Enum):
+    API_KEY = auto()
+    SESSION = auto()
+
+
+class ClaimStatus(Enum):
+    VALID = auto()
+    INACTIVE = auto()
+    EXPIRED = auto()
+
+
+@dataclass(frozen=True)
+class _BaseAttributes:
+    user_role: str
+
+
+@dataclass(frozen=True)
+class ApiKeyAttributes(_BaseAttributes):
+    name: Optional[str] = None
+    description: Optional[str] = None
+
+
+@dataclass(frozen=True)
+class SessionAttributes(_BaseAttributes): ...
+
+
+Attributes: TypeAlias = Union[ApiKeyAttributes, SessionAttributes]
+TokenId: TypeAlias = str
+
+
+@dataclass(frozen=True)
+class Claim:
```

Comment:
this sorta feels a bit like we are shoehorning our own concepts into what in general is scopes and claims. 

https://curity.io/resources/learn/what-are-claims-and-how-they-are-used/

Reply:
It seems to me our usage is compatible. Can you describe why you consider otherwise?

> A claim is statement that a particular entity has a particular property.

> Claims are mapped to a Scope (scope of access) [...] this can be arbitrarily mapped.

Reply:
I guess I've used it mainly to request scopes

> They (claims) are a way of allowing the user to consent to what data it shares with an application or API (discussed in the 
[consent overview](https://curity.io/resources/learn/the-relationship-between-consent-and-claims/)
). They are also a way to enable applications to provide services and perform actions on the user's behalf.

It just feels slightly off to me - but I might be looking at it through the wrong lense.

Reply:
I think you may be referring to [this](https://github.com/encode/starlette/blob/c78c9aac17a4d68e0647252310044502f1b7da71/starlette/authentication.py#L118), which is the first half of the tuple [here](https://github.com/Arize-ai/phoenix/blob/766b1db427763335f6c23ed0147b9349cf846b13/src/phoenix/server/bearer_auth.py#L39), and is yet to be defined. But that separation is part of the intent of this PR, in accordance with the diagram shown in the PR description.

Reply:
To play devil's advocate, these aren't just any JWT claims, these are *our* claims, which do have required fields.

Reply:
Yes, i do agree with that, but i don't know what OAuth looks like yet, so asserts may just be a small price to pay for now

Reply:
FWIW we're planning on a doing an exchange for our tokens on login with an OAuth provider, so OAuth access and refresh tokens won't actually be set in the browser.

Reply:
I think builtin dataclasses has a Field type analogous to pydantic that might work

Reply:
sounds good. i'm gonna merge this one for now to get others unblocked